baseapp/baseapp.go

```diff
@@ -575,6 +617,9 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte, tx sdk.Tx) (result sdk
 	// determined by the GasMeter. We need access to the context to get the gas
 	// meter so we initialize upfront.
 	var gasWanted int64
+	if mode == runTxModeSimulate {
+		return app.simulateTx(txBytes, tx)
```

Comment:
Hmm I'm not sure this is the optimal way to structure this function set, now there seems to be some duplicate code in `simulateTx`.